### DISCUSSION TITLE
PasswordMaker.org-like option for the Password Generator

### DISCUSSION THREAD

**Discussion by rolltidehero** [keepassxreboot/keepassxc#7344] [Ideas]
Can we PLEASE get a PasswordMaker.org-like option for the Password Generator? 

There is source code available for just about every language you can think of, so one of you geniuses may be able to implement this into our beloved KeePassXC. I'll buy you a coffee... or a blunt. I don't judge.

Here is more info on what I am referring to for those who aren't familiar with it...

**PasswordMaker uses three pieces of information:** 
   1. A "master password" -- usually that one, single password you like.
   2. The URL of the website requiring a password. 
   3. The encryption method of your choice.

_Through the magic of one-way hash algorithms, PasswordMaker calculates a message digest, also known as a digital fingerprint, which can be used as your password for the website._ 

_Although one-way hash algorithms have a number of interesting characteristics, the one capitalized by PasswordMaker is that the resulting fingerprint (password) does "not reveal anything about the input that was used to generate it."_  

_In other words, if someone has one or more of your generated passwords, it is computationally infeasible for him to derive your master password or to calculate your other passwords. Computationally infeasible means even computers like this won't help!_

This method is widely used as there are multiple versions available for just about everything:

-      Browser Extensions
  -   [Chrome \ Chromium-based browsers](https://passwordmaker.org/Google_Chrome)
  -   [Firefox](https://passwordmaker.org/Firefox/Mozilla/SeaMonkey/Flock/Netscape/GettingStarted)
  -   [Opera](https://passwordmaker.org/Opera_Widget)
  -   [Bookmarklet](https://passwordmaker.org/Click)

-      Mobile
  -   [iOS](http://itunes.apple.com/us/app/passwordmaker-org-password/id359001896?mt=8) - Available Editions: 1
  -   [Android](https://passwordmaker.org/Android) - Available Editions: 3
  -   [PHP](https://passwordmaker.org/PHP_/_Mobile#Mobile_Edition) - For use on your own web server.

-      Desktop
  -   [Windows](https://passwordmaker.org/Desktop) - Available Editions: 2 
  -   [Mac](https://passwordmaker.org/Mac_Widget) - Available Editions: 2 
  -   [Linux/Python](https://passwordmaker.org/Python) - Available Editions: 2

-      Other
  -   [Online - Original](http://passwordmaker.sourceforge.net/passwordmaker.html) / [Online 2](https://adamghill.com/passwordmaker/) - Available Editions: 2 
  -   [Javascript](https://passwordmaker.org/Javascript) - The Javascript Edition can be used on your local drive or on your own web server. 
  -   [PHP](https://passwordmaker.org/PHP_/_Mobile) - The PHP Edition can be run from the command-line (php cli) or used on a webserver (php cgi).
  -   [Maemo](https://passwordmaker.org/Maemo) - CLI and GUI versions available

-      Other Links
  -  [Screenshots of many of the different editions & extensions](https://passwordmaker.org/Screenshots)
  -  [FAQ about the method and other things](https://passwordmaker.org/F.A.Q.) 
  -  [Wiki where I got all of this information](https://passwordmaker.org/)

### If someone gets my master password, can't he determine all of my generated passwords?
**No. There are ten other variables he would need for each account. They are:**

- URL
- character set
- which of nine hash algorithms was used
- modifier (if any)
- username (if any)
- password length
- password prefix (if any)
- password suffix (if any)
- which of nine l33t-speak levels was used
- when l33t-speak was applied (if at all)

Probably the most interesting of these is character set because it gives you the flexibility to determine precisely which characters can and can't be included in generated passwords.

So... Thoughts? Is this possible? Should I be posting this in browser extension discussions instead? 
I currently use the browser extensions, but it would be so much easier if it was coded into KeePassXC instead.
Regardless, thanks to all you guys & gals for what you do!

**Comment by droidmonkey** [comment]
This is a terrible system and we won't implement it. You can't change your master password without having to change evey single derived password. If your master password is leaked then we can derive every single of your passwords.

**Reply by rolltidehero** [reply]
Not if you don't know the encryption method used. Plus, you would still have to know the username as well as the website.

**Reply by rolltidehero** [reply]
### If someone gets my master password, can't he determine all of my generated passwords?
**No. There are ten other variables he would need for each account. They are:**

- URL
- character set
- which of nine hash algorithms was used
- modifier (if any)
- username (if any)
- password length
- password prefix (if any)
- password suffix (if any)
- which of nine l33t-speak levels was used
- when l33t-speak was applied (if at all)

Probably the most interesting of these is character set because it gives you the flexibility to determine precisely which characters can and can't be included in generated passwords.

**Reply by droidmonkey** [reply]
And where do you store those ten variables? In keepassxc? Why aren't you just using a random password and eliminate all that nonsense. It makes zero sense and it will never be in our program.

**Reply by StarBoyzz** [reply]
Hell, just the custom prefix and\or suffix option would make it nearly impossible to steal whether they have your master password or not.
